### PR TITLE
Shakemap importer (reclassify raster)

### PR DESCRIPTION
**Pull request by myarjunar**
### What does it fix ?
* Ticket #3874 
* Funded by : DFAT
* Description : half decimal on either side of the mmi level get assigned eg. 0.5 - 1.5 get reclassed as 1

### CODE REVIEW

File: safe/gui/tools/shake_grid/shake_grid.py [hunk 1]
```diff
@@ -841,14 +848,21 @@ def create_keyword_file(self, algorithm):
         """
         keyword_io = KeywordIO()
 
+        classes = {}
+        for item in earthquake_mmi_scale['classes']:
+            classes[item['key']] = [item['numeric_default_min'],
```
Comment (timlinux) [inline safe/gui/tools/shake_grid/shake_grid.py]: pull left

File: safe/gis/raster/reclassify.py [hunk 2]
```diff
@@ -94,7 +98,10 @@ def reclassify(layer, exposure_key=None, callback=None):
         ranges[hazard_class['value']] = thresholds[hazard_class['key']]
         value_map[hazard_class['key']] = [hazard_class['value']]
 
-    output_raster = unique_filename(suffix='.tiff', dir=temp_dir())
+    if overwrite_input:
+        output_raster = layer.publicSource()
```
Comment (Gustry) [inline safe/gis/raster/reclassify.py]: I think you need to use 'source'.  'publicSource' will have some stars for instance if there is a password.

File: safe/gui/tools/shake_grid/shake_grid.py [hunk 3]
```diff
@@ -841,14 +848,21 @@ def create_keyword_file(self, algorithm):
...
-            'hazard': 'earthquake',
-            'hazard_category': 'single_event',
-            'keyword_version': '3.3',
-            'layer_geometry': 'raster',
-            'layer_mode': 'continuous',
-            'layer_purpose': 'hazard',
-            'continuous_hazard_unit': 'mmi',
+            'hazard': hazard_earthquake['key'],
+            'hazard_category': hazard_category_single_event['key'],
+            'keyword_version': '3.3',  # will be replaced later
```
Comment (Gustry) [inline safe/gui/tools/shake_grid/shake_grid.py]: Can you use definition for the version ?

File: safe/utilities/styling.py [hunk 4]
```diff
@@ -573,3 +575,40 @@ def mmi_colour(mmi_value):
                 '#D00', '#800', '#400']
     rgb = rgb_list[int(mmi_value)]
     return rgb
+
+
+def mmi_ramp_roman(raster_layer):
+    """Generate an mmi ramp using range of 1-10 on roman.
+
```
Comment (timlinux) [inline safe/utilities/styling.py]: please annotate new methods with :versionadded:

**Pull request merged**